charger/switchsocket.go

```diff
@@ -58,13 +59,21 @@ func NewSwitchSocketFromConfig(ctx context.Context, other map[string]interface{}
 		}
 	}
 
+	var soc func() (float64, error)
+	if cc.Soc != nil {
+		soc, err = provider.NewFloatGetterFromConfig(ctx, *cc.Energy)
```

Comment:
@andig Mir fällt gerade in der Mittagspause auf, dass hier noch ein kleiner Fehler ist:
![image](https://github.com/user-attachments/assets/e8f950f5-0111-404d-97c1-10fa572aaeca)

Reply:
Lange behoben 🤫

Reply:
Btw: Frohes Neues Jahr 2025!